code/modules/mob/living/status_procs.dm

```diff
@@ -606,3 +606,59 @@
 		add_movespeed_modifier(MOVESPEED_ID_COLD, TRUE, 0, NONE, TRUE, 2)
 	else if(old_temperature < species.cold_level_1)
 		remove_movespeed_modifier(MOVESPEED_ID_COLD)
+
+////////////////////////////// STAGGER ////////////////////////////////////
+
+/mob/living/proc/IsStaggered() //If we're staggered
+	return stagger
+
+/mob/living/proc/handle_stagger()
+	if(stagger)
+		adjust_stagger(-1)
+	return stagger
+
+/mob/living/proc/adjust_stagger(amount, ignore_canstun = FALSE)
+	if(SEND_SIGNAL(src, COMSIG_LIVING_STATUS_STUN, amount, ignore_canstun) & COMPONENT_NO_STUN) //Stun immunity also provides immunity to its lesser cousin stagger
+		return
+	stagger = max(stagger + amount,0)
+	return stagger
+
+/mob/living/carbon/xenomorph/adjust_stagger(amount)
+	if(is_charging >= CHARGE_ON) //If we're charging we don't accumulate more stagger stacks.
+		return FALSE
+	return ..()
```

Comment:
Why is this not in the xeno files

Reply:
Prefer to aggregate all references to stagger and slow here

Reply:
move it to xeno files